### PR TITLE
ci: fix pinning script

### LINKED ISSUE #14500: dev/tools/pin-ci.sh is broken
The script expects repositories to be described as `foo_CI_REF:=bar`. This is no longer the case since #13603. Obviously, this prevents pinning the repositories for the 8.14 branch.

As a side note, the fix should use a slash-free regex, if any, as the branch name for Perennial is `coq/tested`.

### PR DESCRIPTION
**Pull request by gares**
Fix #14500

### CODE REVIEW

File: dev/tools/pin-ci.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ process_development() {
   echo
   if [[ $REPLY =~ ^[Yy]$ ]]; then
     # use -i.bak to be compatible with MacOS; see, e.g., https://stackoverflow.com/a/7573438/377022
-    sed -i.bak -e "s/$BRANCH_VAR:=$BRANCH/$BRANCH_VAR:=$HASH/" $OVERLAYS
+    sed -i.bak -E "s?project +$DEV +.*?project $DEV \"$REPO\" \"$HASH\"?" $OVERLAYS
```
Comment (SkySkimmer) [inline dev/tools/pin-ci.sh:34]: How about using `|` as a separator instead? `?` feels like it should mean "0 or 1 occurences"

Comment (SkySkimmer) [reply dev/tools/pin-ci.sh:34]: also if you use single quotes in the result you don't have to backslash
```suggestion
    sed -i.bak -E "s?project +$DEV +.*?project $DEV '$REPO' '$HASH'?" $OVERLAYS
```

**Review by silene — APPROVED** [review]
I share the stylistic concerns of @SkySkimmer. Other than that, it looks good, thanks a lot. (I still need to test it, but I suppose you already have done so.)

File: dev/tools/pin-ci.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ process_development() {
   echo
   if [[ $REPLY =~ ^[Yy]$ ]]; then
     # use -i.bak to be compatible with MacOS; see, e.g., https://stackoverflow.com/a/7573438/377022
-    sed -i.bak -e "s/$BRANCH_VAR:=$BRANCH/$BRANCH_VAR:=$HASH/" $OVERLAYS
+    sed -i.bak -E "s?project +$DEV +.*?project $DEV '$REPO' '$HASH'?" $OVERLAYS
```
Comment (gares) [inline dev/tools/pin-ci.sh:34]:
```suggestion
    sed -i.bak -E "s|project +$DEV +.*|project $DEV '$REPO' '$HASH'|" $OVERLAYS
```

**Comment by gares** [comment]
Here what it does:
```diff
diff --git a/dev/ci/ci-basic-overlay.sh b/dev/ci/ci-basic-overlay.sh
index 73b9ca7a48..6fe8671286 100755
--- a/dev/ci/ci-basic-overlay.sh
+++ b/dev/ci/ci-basic-overlay.sh
@@ -42,32 +42,32 @@ function project {
 ########################################################################
 # MathComp
 ########################################################################
-project mathcomp "https://github.com/math-comp/math-comp" "master"
+project mathcomp 'https://github.com/math-comp/math-comp' '10816c7d830a568d732c749831466ba8830ffe49'
 
 project fourcolor "https://github.com/math-comp/fourcolor" "master"
 
-project oddorder "https://github.com/math-comp/odd-order" "master"
+project oddorder 'https://github.com/math-comp/odd-order' '91e3b9a75d12d142568985566e652988189a2ff4'
 
 project mczify "https://github.com/math-comp/mczify" "master"
 
 ########################################################################
 # UniMath
 ########################################################################
-project unimath "https://github.com/UniMath/UniMath" "master"
+project unimath 'https://github.com/UniMath/UniMath' 'd0e7d748fb5cee00472c08abe1835022910ec7ff'
 
 ########################################################################
 # Unicoq + Mtac2
 ########################################################################
 project unicoq "https://github.com/unicoq/unicoq" "master"
 
-project mtac2 "https://github.com/Mtac2/Mtac2" "master"
+project mtac2 'https://github.com/Mtac2/Mtac2' 'c27161ba0e9c750976f343cae3245708580c8079'
 
 ########################################################################
 # Mathclasses + Corn
 ########################################################################
 project math_classes "https://github.com/coq-community/math-classes" "master"
 
-project corn "https://github.com/coq-community/corn" "master"
+project corn 'https://github.com/coq-community/corn' 'c366d3f01ec1812b145117a4da940518b092d3a6'
 
 ########################################################################
 # Iris
@@ -81,7 +81,7 @@ project iris "https://gitlab.mpi-sws.org/iris/iris" ""
 
 project autosubst "https://github.com/coq-community/autosubst" "master"
 
-project iris_examples "https://gitlab.mpi-sws.org/iris/examples" "master"
+project iris_examples 'https://gitlab.mpi-sws.org/iris/examples' 'ee6a36b3e803ebec465bf44bb5bcbf9b19fe6b62'
 
 ########################################################################
 # HoTT
@@ -91,7 +91,7 @@ project hott "https://github.com/HoTT/HoTT" "master"
 ########################################################################
 # CoqHammer
 ########################################################################
-project coqhammer "https://github.com/lukaszcz/coqhammer" "master"
+project coqhammer 'https://github.com/lukaszcz/coqhammer' '1079879790f671389e282a3606c0eda996b6c16a'
 
 ########################################################################
 # GeoCoq
@@ -101,7 +101,7 @@ project geocoq "https://github.com/GeoCoq/GeoCoq" "master"
 ########################################################################
 # Flocq
 ########################################################################
-project flocq "https://gitlab.inria.fr/flocq/flocq" "flocq-3"
+project flocq 'https://gitlab.inria.fr/flocq/flocq' '239727fc0669d76ee60a45efb176bda2fe261f58'
 
 ########################################################################
 # coq-performance-tests
@@ -111,7 +111,7 @@ project coq_performance_tests "https://github.com/coq-community/coq-performance-
 ########################################################################
 # coq-tools
 ########################################################################
-project coq_tools "https://github.com/JasonGross/coq-tools" "master"
+project coq_tools 'https://github.com/JasonGross/coq-tools' '366c217bf1d7b158d0c1721d7e8f254f025f89a4'
 
 ########################################################################
 # Coquelicot
@@ -121,7 +121,7 @@ project coquelicot "https://gitlab.inria.fr/coquelicot/coquelicot" "master"
 ########################################################################
 # Gappa stand alone tool
 ########################################################################
-project gappa_tool "https://gitlab.inria.fr/gappa/gappa" "master"
+project gappa_tool 'https://gitlab.inria.fr/gappa/gappa' '6c97a36257369d89ff32c9877c0e83681bfd3df9'
 
 ########################################################################
 # Gappa plugin
@@ -131,7 +131,7 @@ project gappa "https://gitlab.inria.fr/gappa/coq" "master"
 ########################################################################
 # CompCert
 ########################################################################
-project compcert "https://github.com/AbsInt/CompCert" "master"
+project compcert 'https://github.com/AbsInt/CompCert' 'a0ad5ff6f9c7603610a7448935b36c9ed22c6435'
 
 ########################################################################
 # VST
@@ -142,7 +142,7 @@ project vst "https://github.com/PrincetonUniversity/VST" "compcert3.9"
 ########################################################################
 # cross-crypto
 ########################################################################
-project cross_crypto "https://github.com/mit-plv/cross-crypto" "master"
+project cross_crypto 'https://github.com/mit-plv/cross-crypto' 'c7cc46de53cf6b2363631a3b17b2ab12ce42e2f8'
 
 ########################################################################
 # rewriter
@@ -152,7 +152,7 @@ project rewriter "https://github.com/mit-plv/rewriter" "master"
 ########################################################################
 # fiat_parsers
 ########################################################################
-project fiat_parsers "https://github.com/mit-plv/fiat" "master"
+project fiat_parsers 'https://github.com/mit-plv/fiat' '490532505f20d3c93e0936f5770d2358c6effaaf'
 
 ########################################################################
 # fiat_crypto
@@ -162,7 +162,7 @@ project fiat_crypto "https://github.com/mit-plv/fiat-crypto" "master"
 ########################################################################
 # fiat_crypto_legacy
 ########################################################################
-project fiat_crypto_legacy "https://github.com/mit-plv/fiat-crypto" "sp2019latest"
+project fiat_crypto_legacy 'https://github.com/mit-plv/fiat-crypto' '093ea19d246a3a79d019b5a78606559b261cf7c1'
 
 ########################################################################
 # coq_dpdgraph
@@ -172,7 +172,7 @@ project coq_dpdgraph "https://github.com/Karmaki/coq-dpdgraph" "coq-master"
 ########################################################################
 # CoLoR
 ########################################################################
-project color "https://github.com/fblanqui/color" "master"
+project color 'https://github.com/fblanqui/color' '1e7b26553c1ca94c787ad5a1b938acabb8d47f2f'
 
 ########################################################################
 # TLC
@@ -182,7 +182,7 @@ project tlc "https://github.com/charguer/tlc" "master-for-coq-ci"
 ########################################################################
 # Bignums
 ########################################################################
-project bignums "https://github.com/coq/bignums" "master"
+project bignums 'https://github.com/coq/bignums' 'a419f649a7a34e28049f3f18a5ee4068c2cd7fff'
 
 ########################################################################
 # coqprime
@@ -192,7 +192,7 @@ project coqprime "https://github.com/thery/coqprime" "master"
 ########################################################################
 # bbv
 ########################################################################
-project bbv "https://github.com/mit-plv/bbv" "master"
+project bbv 'https://github.com/mit-plv/bbv' '1386458cdf0b87bd9deaa2cdc47d3d732f96f5b6'
 
 ########################################################################
 # bedrock2
@@ -202,14 +202,14 @@ project bedrock2 "https://github.com/mit-plv/bedrock2" "tested"
 ########################################################################
 # Equations
 ########################################################################
-project equations "https://github.com/mattam82/Coq-Equations" "master"
+project equations 'https://github.com/mattam82/Coq-Equations' '3b48ad15c64fe1f3ec81978679960f265a7877b5'
 
 ########################################################################
 # Elpi + Hierarchy Builder
 ########################################################################
 project elpi "https://github.com/LPCIC/coq-elpi" "coq-master"
 
-project hierarchy_builder "https://github.com/math-comp/hierarchy-builder" "coq-master"
+project hierarchy_builder 'https://github.com/math-comp/hierarchy-builder' 'bd589b63bf48cf9a00812bdaa0c9a8192e78b83b'
 
 ########################################################################
 # Engine-Bench
@@ -219,7 +219,7 @@ project engine_bench "https://github.com/mit-plv/engine-bench" "master"
 ########################################################################
 # fcsl-pcm
 ########################################################################
-project fcsl_pcm "https://github.com/imdea-software/fcsl-pcm" "master"
+project fcsl_pcm 'https://github.com/imdea-software/fcsl-pcm' 'cbd73130489169d1bb236a0888fceacaa77f0d70'
 
 ########################################################################
 # ext-lib
@@ -229,7 +229,7 @@ project ext_lib "https://github.com/coq-community/coq-ext-lib" "master"
 ########################################################################
 # simple-io
 ########################################################################
-project simple_io "https://github.com/Lysxia/coq-simple-io" "master"
+project simple_io 'https://github.com/Lysxia/coq-simple-io' 'f7536c629dd02883abb5a2d64865becafed3af5b'
 
 ########################################################################
 # quickchick
@@ -239,7 +239,7 @@ project quickchick "https://github.com/QuickChick/QuickChick" "master"
 ########################################################################
 # reduction-effects
 ########################################################################
-project reduction_effects "https://github.com/coq-community/reduction-effects" "master"
+project reduction_effects 'https://github.com/coq-community/reduction-effects' '7cf41546f2d15ea3a4eeeb68695b288db2d9e09a'
 
 ########################################################################
 # menhirlib
@@ -255,7 +255,7 @@ project aac_tactics "https://github.com/coq-community/aac-tactics" "master"
 ########################################################################
 # paramcoq
 ########################################################################
-project paramcoq "https://github.com/coq-community/paramcoq" "master"
+project paramcoq 'https://github.com/coq-community/paramcoq' '1d9409b9eb56cb3f60e176e1f9194d6c4e09afef'
 
 ########################################################################
 # relation_algebra
@@ -265,15 +265,15 @@ project relation_algebra "https://github.com/damien-pous/relation-algebra" "mast
 ########################################################################
 # StructTact + InfSeqExt + Cheerios + Verdi + Verdi Raft
 ########################################################################
-project struct_tact "https://github.com/uwplse/StructTact" "master"
+project struct_tact 'https://github.com/uwplse/StructTact' '179bd5312e9d8b63fc3f4071c628cddfc496d741'
 
 project inf_seq_ext "https://github.com/DistributedComponents/InfSeqExt" "master"
 
-project cheerios "https://github.com/uwplse/cheerios" "master"
+project cheerios 'https://github.com/uwplse/cheerios' '9c7f66e57b91f706d70afa8ed99d64ed98ab367d'
 
 project verdi "https://github.com/uwplse/verdi" "master"
 
-project verdi_raft "https://github.com/uwplse/verdi-raft" "master"
+project verdi_raft 'https://github.com/uwplse/verdi-raft' 'dbf3084f8a741d56b5d05a87e39705329753bb6e'
 
 ########################################################################
 # stdlib2
@@ -283,7 +283,7 @@ project stdlib2 "https://github.com/coq/stdlib2" "master"
 ########################################################################
 # argosy
 ########################################################################
-project argosy "https://github.com/mit-pdos/argosy" "master"
+project argosy 'https://github.com/mit-pdos/argosy' '63dd6e6d046e3dde072ae1b6862639a8f62b1213'
 
 ########################################################################
 # perennial
@@ -293,7 +293,7 @@ project perennial "https://github.com/mit-pdos/perennial" "coq/tested"
 ########################################################################
 # metacoq
 ########################################################################
-project metacoq "https://github.com/MetaCoq/metacoq" "master"
+project metacoq 'https://github.com/MetaCoq/metacoq' '1407f12779cadd7412f7a16607d69788590715a8'
 
 ########################################################################
 # SF suite
@@ -303,7 +303,7 @@ project sf "https://github.com/DeepSpec/sf" "master"
 ########################################################################
 # Coqtail
 ########################################################################
-project coqtail "https://github.com/whonore/Coqtail" "master"
+project coqtail 'https://github.com/whonore/Coqtail' '2f1c81a86342e333a916a1bc7db5218cf115a191'
 
 ########################################################################
 # Deriving
@@ -313,4 +313,4 @@ project deriving "https://github.com/arthuraa/deriving" "master"
 ########################################################################
 # VsCoq
 ########################################################################
-project vscoq "https://github.com/maximedenes/vscoq" document-manager
+project vscoq 'https://github.com/maximedenes/vscoq' 'f6f61c52d7e01b18dd4fea19604ee0a2bf1be22c'
```

There are a fix skips, but they seem unrelated to this change

**Comment by gares** [comment]
Hum, something is wrong, since I need to run it a few times before all projects get pinned...

**Comment by silene** [comment]
I just tried it on 8.14. It went mostly right. I just need to have look at menhirlib to see if the failure is expected.

**Comment by silene** [comment]
By the way, I am changing the milestone because the simple existence of the script is sufficient. I don't really need it to be backported.

**Comment by silene** [comment]
@gares Do you still think that there is something wrong which prevents merging?

**Comment by gares** [comment]
No, I think I lack sleep ;-)

**Comment by silene** [comment]
@coqbot merge now

**Pull request merged**